### PR TITLE
Refactor navbar to use PrimeNG menubar

### LINKED ISSUE #320: PrimeNG: Refactor navbar to use PrimeNG menubar
The current nav bar should be refactored to utilize PrimeNG components.  I think the PrimeNG [menubar](https://primefaces.org/primeng/showcase/#/menubar) would be a good place to start but I'm open to other creative ideas.

Furthermore, as part of this issue, please fix the alignment of the loading bar.  Currently, it's off center.  Please see image below:

Screen Shot 2020-10-02 at 9 18 28 AM

### PR DESCRIPTION
**Pull request by JoelLau**
# Pull Request

Refactors the navbar component to use PrimeNG's menubar component instead of `<nav>` elements

Fixes #320 

## Type of change

- [ ] A new feature
- [ ] A bug fix
- [ ] Documentation only changes
- [ ] Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [x] A code change that neither fixes a bug nor adds a feature
- [ ] A code change that improves performance
- [ ] Adding missing tests or correcting existing tests
- [ ] Changes that affect the build system or external dependencies
- [ ] Changes to our CI configuration files and scripts
- [ ] Other changes that don't modify src or test files

```release-note

```

---

## Checklist

- [x] My code follows the [contributing](../CONTRIBUTING.md) guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Whamo12** [comment]
@JoelLau please open this pull request against the `develop` branch please.

**Comment by JoelLau** [comment]
right, my bad

File: frontend/src/app/navbar/navbar.component.html [hunk 1]
```diff
@@ -1,22 +1,10 @@
...
-    </div>
-  </div>
-  <div class="nav-item dropdown" *ngIf="loggedIn$ | async as loggedIn">
-    <a class="nav-link dropdown-toggle" id="navbarDropdown" role="button" data-toggle="dropdown" aria-haspopup="true"
-      aria-expanded="false">
-      Menu
+<div *ngIf="loaderService.isLoading | async" role="status">
+  <p-progressSpinner [style]="{width: '75px', height: '75px'}"></p-progressSpinner>
+</div>
+<p-menubar [model]="menuItems" *ngIf="loggedIn$ | async as loggedIn">
```
Comment (Whamo12) [inline frontend/src/app/navbar/navbar.component.html:4]: Screen Shot 2020-10-18 at 10 34 31 AM
This conditional will completely hide the menu bar when the user logs out.  Instead, we need it so the menu items are hidden when the user is logged out.

Screen Shot 2020-10-18 at 10 24 36 AM

File: frontend/src/app/navbar/navbar.component.html [hunk 2]
```diff
@@ -1,22 +1,10 @@
...
-  <div class="mx-auto">
-    <div *ngIf="loaderService.isLoading | async" role="status">
-      <p-progressSpinner [style]="{width: '75px', height: '75px'}"></p-progressSpinner>
-    </div>
-  </div>
-  <div class="nav-item dropdown" *ngIf="loggedIn$ | async as loggedIn">
-    <a class="nav-link dropdown-toggle" id="navbarDropdown" role="button" data-toggle="dropdown" aria-haspopup="true"
-      aria-expanded="false">
-      Menu
+<div *ngIf="loaderService.isLoading | async" role="status">
```
Comment (Whamo12) [inline frontend/src/app/navbar/navbar.component.html:1]: Currently, the loading spinner is rendering on top of the loading spinner causing it to push the menu bar down with every API request.  The loading spinner should display in the center of the menu bar and should not cause the menu bar to move.

File: frontend/src/app/navbar/navbar.component.ts [hunk 3]
```diff
@@ -11,14 +12,45 @@ import { GlobalManagerService } from '../global-manager.service';
...
+        icon: 'pi pi-fw pi-sliders-h',
+        routerLink: ['settings']
+      },
+      {
+        separator: true
+      },
+      {
+        label: 'Logout',
+        icon: 'pi pi-fw pi-power-off',
+        command: this.logout
```
Comment (Whamo12) [inline frontend/src/app/navbar/navbar.component.ts:50]: This is throwing an error that `this.authService` is undefined.



Screen Shot 2020-10-18 at 10 36 00 AM
 I believe the syntax for `command` should be

```
command: () => {
   this.authService.logout()
}
```
Furthermore, we need to make sure that the user is redirected to the login screen.

File: frontend/src/app/navbar/navbar.component.html (hunk 1 above)
Comment (Whamo12) [inline frontend/src/app/navbar/navbar.component.html:4]: Add space between the menu bar and the body web components.  Currently, they are attached.

Screen Shot 2020-10-18 at 10 46 35 AM

**Review by Whamo12 — CHANGES_REQUESTED** [review]

**Review by Whamo12 — CHANGES_REQUESTED** [review]
The PR was requested for review even though no updates were made to fix the existing issues.

File: frontend/src/app/navbar/navbar.component.html (hunk 2 above)
Comment (JoelLau) [reply frontend/src/app/navbar/navbar.component.html:1]: hi @Whamo12, what do you think about moving the loading bar to cover the whole page using something like [ng-block-ui](https://github.com/kuuurt13/ng-block-ui)?

**Pull request closed without merging**